blockchain/chain_sync/src/network_context.rs

```diff
@@ -62,6 +65,47 @@ impl<DB: Clone> Clone for SyncNetworkContext<DB> {
     }
 }
 
+type RaceBatchFuture<T> = Pin<Box<dyn Future<Output = Result<T, String>> + Send>>;
```

Comment:
Let's have some documentation and possibly a set of unit tests for this. It's not clear from a first look of this for someone unfamiliar with synchronization primitives.

Reply:
Added some unit tests!